aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -151,6 +152,28 @@ Tensor rsub(const Tensor& self, const Tensor& other, Scalar alpha) {
   return native::sub(other, self, alpha);
 }
 
+
+Tensor& atan2_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  auto device_res = result.type().device_type();
+  auto device1 = self.type().device_type();
+  TORCH_CHECK(device_res == device1, "out and input1 must have the same device. out: ", device_res, " input1: ", device1);
```

Comment:
does it make any sense to move the checking to TensorIterator itself?

Reply:
It is already inside of thr TensorIterator

Reply:
Fixed